app/connectors/launchpadgateway/exchangeobjects/in/reviewed/ReviewedCallbackRequest.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.launchpadgateway.exchangeobjects.in.reviewed
+
+import connectors.launchpadgateway.exchangeobjects.in.reviewed.ReviewedCallbackRequest.LaunchpadQuestionIsUnscoredException
+import org.joda.time.{DateTime, LocalDate}
+import play.api.libs.json.Json
+import reactivemongo.bson.{BSONDocument, BSONHandler, Macros}
+
+case class ReviewedCallbackRequest(
+  received: DateTime,
+  candidateId: String,
+  customCandidateId: String,
+  interviewId: Int,
+  customInterviewId: Option[String],
+  customInviteId: String,
+  deadline: LocalDate,
+  reviews: ReviewSectionRequest) {
+
+  def calculateTotalScore(): Double = {
+    val reviewers = reviews.reviewers
+    val latestReviewer = reviewers.reviewer3.getOrElse(reviewers.reviewer2.getOrElse(reviewers.reviewer1))
+
+    def throwUnscored(questionNo: Int): Double = throw LaunchpadQuestionIsUnscoredException(s"$questionNo is unscored, score cannot be calculated")
+
+    def scoreForQuestion(questionNo: Int, question: ReviewSectionQuestionRequest) =
+      question.reviewCriteria1.score.getOrElse(throwUnscored(questionNo)) + question.reviewCriteria2.score.getOrElse(throwUnscored(questionNo))
+
+    scoreForQuestion(1, latestReviewer.question1) +
+    scoreForQuestion(2, latestReviewer.question2) +
+    scoreForQuestion(3, latestReviewer.question3) +
+    scoreForQuestion(4, latestReviewer.question4) +
+    scoreForQuestion(5, latestReviewer.question5) +
+    scoreForQuestion(6, latestReviewer.question6) +
+    scoreForQuestion(7, latestReviewer.question7) +
+    scoreForQuestion(8, latestReviewer.question8)
+  }
+}
```

Comment:
@ollyw I think this is good enough, it will let us fetch a test (with its callbacks) and do dynamic scoring.

I've abandoned the scoring-as-callbacks-come-in idea because in some situations i don't think we want to calculate a score (see LaunchpadQuestionIsUnscoredException above). I think that when the score processor comes to a user it should call this method and if it gets the exception carry on to come back to this user later.

The reviewer team will proactively contact the candidate via Launchpad support who will prompt them for resets/to retake questions that cannot be scored, the assumption is that ultimately no one will have an unscored question - and if they do we won't process them. Need to check this with Mike, there's a ticket on the board about it.